### PR TITLE
Hotfix/pagination

### PR DESCRIPTION
**Pull request by AltysS**

### CODE REVIEW

**Comment by AltysS** [comment]
Sorry for the unclear commits have an issue with VS code. I`ll write all the changes here:
1) Comment the fetch functions at 2 files - CatalogueFilterDuration, CatalogueMainFilter. The logic here must to be changes as my thunk function fetches only 5 items now instead of all catalogue. 

2) Add new thunk function to the CardContainer file. Now it fetching popular tours with thunk function.

3) CataloguePage - rewrite pagination logic, change fetch function with thunk function, delete unnecessary pagination code.

4) Comment a favorites page fetch function as it breaks pagination logic. Favorites page is not working now and has to be fixed soon by me.

5) Change catalogue slice. Rewrite all functions with create async thunk.

File: client/src/store/slices/catalogueSlice/catalogueSlice.js [hunk 1]
```diff
@@ -1,52 +1,76 @@
 /* eslint-disable no-return-assign */
 /* eslint-disable no-param-reassign */
-import { createSlice } from '@reduxjs/toolkit';
-import axios from 'axios';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axiosConfig from '../../../axiosConfig';
+
+export const fetchCatalogue = createAsyncThunk('products/fetchCatalogue', async () => {});
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/slices/catalogueSlice/catalogueSlice.js:6]: ???

File: client/src/store/slices/catalogueSlice/catalogueSlice.js [hunk 2]
```diff
@@ -1,52 +1,76 @@
...
+
+export const fetchCatalogueProducts = createAsyncThunk(
+  'products/fetchCatalogueProducts',
+  async (page, { rejectWithValue }) => {
+    try {
+      const { data, status } = await axiosConfig.get(`products/filter?perPage=5&startPage=${page}`);
+      if (status >= 200 && status <= 300) {
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/slices/catalogueSlice/catalogueSlice.js]: u dont need that check as createAsyncThunk function checks by itself and returns fulfilled if status is between 200 - 300

File: client/src/store/slices/catalogueSlice/catalogueSlice.js [hunk 3]
```diff
@@ -1,52 +1,76 @@
...
+        return data;
+      }
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
+export const fetchPopularProducts = createAsyncThunk('products/isPopular', async (_, {rejectWithValue}) => {
+  try {
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/slices/catalogueSlice/catalogueSlice.js]:
```suggestion
export const fetchPopularProducts = createAsyncThunk('products/fetchPopularProducts ', async (_, {rejectWithValue}) => {
  try {
```

File: client/src/store/slices/catalogueSlice/catalogueSlice.js [hunk 4]
```diff
@@ -1,52 +1,76 @@
...
+    const { status, data } = await axiosConfig('/products/filter?isPopular=true');
+    if (status >= 200 && status <= 300) {
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/slices/catalogueSlice/catalogueSlice.js]: the same, check not needed

File: client/src/store/slices/catalogueSlice/catalogueSlice.js [hunk 5]
```diff
@@ -1,52 +1,76 @@
...
+    currentPage: 1,
   },
   reducers: {
-    setProducts: (state, action) => {
-      state.products = action.payload;
-    },
-
     setPopular: (state, action) => {
       state.popular = action.payload;
     },
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/store/slices/catalogueSlice/catalogueSlice.js]: i guess this reducer function is not needed anymore, please delete.

File: client/src/store/slices/catalogueSlice/catalogueSlice.js (hunk 1 above)
Comment (AltysS) [reply client/src/store/slices/catalogueSlice/catalogueSlice.js:6]: this has to be used in the future to fetch all catalogue products. For e.g., once the filter has to be used.

File: client/src/store/slices/catalogueSlice/catalogueSlice.js (hunk 2 above)
Comment (AltysS) [reply client/src/store/slices/catalogueSlice/catalogueSlice.js]: Noted well. Thank you. solved

File: client/src/store/slices/catalogueSlice/catalogueSlice.js (hunk 4 above)
Comment (AltysS) [reply client/src/store/slices/catalogueSlice/catalogueSlice.js]: solved

File: client/src/store/slices/catalogueSlice/catalogueSlice.js (hunk 3 above)
Comment (AltysS) [reply client/src/store/slices/catalogueSlice/catalogueSlice.js]: corrected

File: client/src/store/slices/catalogueSlice/catalogueSlice.js (hunk 5 above)
Comment (AltysS) [reply client/src/store/slices/catalogueSlice/catalogueSlice.js]: solved

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]

**Review by HIMDanny — APPROVED** [review]

**Pull request merged**